README.md

```diff
@@ -5,12 +5,65 @@
 
 `amphp/aerys-session` is a non-blocking session handler for use with [`amphp/aerys`](https://github.com/amphp/aerys).
 
-**Required PHP Version**
+## Requirements
 
-- PHP 7+
+- PHP 7
 
-**Installation**
+## Installation
 
 ```bash
 $ composer require amphp/aerys-session
 ```
+
+## Registering the Session Middleware
+
+```php
+$redis = new Amp\Redis\Client("tcp://localhost:6379");
+$mutex = new Amp\Redis\Mutex("tcp://localhost:6379");
+
+$router->use(
+    Aerys\session(new Aerys\Session\Redis($redis, $mutex)
+);
+```
+
+## Using the Session
+
+```php
+public function respondToRequest(Request $request, Response $response) {
+    $session = new Aerys\Session($request);
+
+    // you need to read the session before you can access the data from it
+    yield $session->read();
+
+    if (!$session->has("user")) {
+        $response
+            ->setStatus(401)
+            ->end("Unauthorised");
+
+        return;
+    }
+
+    $user = $session->get("user");
+
+    // ...
+
+    // you need to open the session for writing before you can write to it
+    yield $session->open();
+
+    $session->set("token", $token);
+}
+```
+
+## Documentation
+
+- [Official Documentation](http://amphp.org/aerys)
+- [Getting Started with Aerys](http://blog.kelunik.com/2015/10/21/getting-started-with-aerys.html)
+- [Getting Started with Aerys WebSockets](http://blog.kelunik.com/2015/10/20/getting-started-with-aerys-websockets.html)
```

Comment:
they show the server config around which this session code will be. I can take em out if you like.